docker/docker-entrypoint.sh

```diff
@@ -1,20 +1,100 @@
-#!/bin/bash
+#!/bin/sh
+
+function check_migrate() {
+  : "${PROMGEN_DISABLE_MIGRATE:=0}"
+  # These env vars are used as parameters when registering a new shard.
+  # Set defaults:
+  : "${PROMGEN_REGISTER_SHARD:=docker-demo}"
+  : "${PROMGEN_REGISTER_HOST:=prometheus}"
+  : "${PROMGEN_REGISTER_PORT:=9090}"
+
+  if [ ! "${PROMGEN_DISABLE_MIGRATE}" = "1" ]; then
+    set -x
+
+    promgen migrate
+    promgen register "${PROMGEN_REGISTER_SHARD}" "${PROMGEN_REGISTER_HOST}" "${PROMGEN_REGISTER_PORT}"
+
+    set +x
+  fi
+}
+
+function copy_static() {
+  if [ ! -f /srv/promgen/www/static/.present ]; then
```

Comment:
Just reading the code, I'm not entirely sure the purpose for having both /srv/promgen/www and /usr/src/app/static. Is this so that one can mount the directory and then serve the files from nginx/apache or for some other reason?
I think a comment either in the file or the docker instructions will be necessary for clarification :)

Reply:
Yes, the reason is the docker volume management: Every time the container is created, a new volume is allocated and mounted at the path `/srv/promgen/www`. This means all content in the directory (e.g. created in the Dockerfile) will be invisible.
So instead, on the start of the web app, the files are copied across, if they don't exist yet. The `.present` file represents a minor security consideration: When not forbidden by the web server, the file will be accessible by anyone with HTTP access. I decided to only run `collectstatic` in the Dockerfile, just in case it involves some computationally complex work.

Other containers can then mount the volume (e.g. with the `--volumes-from` flag, by mounting a custom volume in both containers or by using the bind volumes).

Reply:
One consideration instead of adding a new file (.present) perhaps can just test for a file that we expect to be there, `/srv/promgen/www/static/js/promgen.js` otherwise, seems like a reasonable explanation :)

Reply:
That sounds good :+1: